### PR TITLE
[AN-5651][AN-5689] Fix tracking

### PR DESCRIPTION
**Pull request by JPonte**
#### APK
[Download build #9555](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9555/artifact/build/artifact/wire-dev-PR1166-9555.apk)
[Download build #9559](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9559/artifact/build/artifact/wire-dev-PR1166-9559.apk)
[Download build #9562](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9562/artifact/build/artifact/wire-dev-PR1166-9562.apk)
[Download build #9566](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9566/artifact/build/artifact/wire-dev-PR1166-9566.apk)
[Download build #9577](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9577/artifact/build/artifact/wire-dev-PR1166-9577.apk)
[Download build #9578](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9578/artifact/build/artifact/wire-dev-PR1166-9578.apk)
[Download build #9589](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9589/artifact/build/artifact/wire-dev-PR1166-9589.apk)
[Download build #9591](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9591/artifact/build/artifact/wire-dev-PR1166-9591.apk)
[Download build #9593](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9593/artifact/build/artifact/wire-dev-PR1166-9593.apk)
[Download build #9598](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9598/artifact/build/artifact/wire-dev-PR1166-9598.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/views/SwitchPreference.scala [hunk 1]
```diff
@@ -44,33 +44,27 @@ class SwitchPreference(context: Context, attrs: AttributeSet, style: Int) extend
     context.getTheme.obtainStyledAttributes(attrs, R.styleable.SwitchPreference, 0, 0)
 
   val keyAttr = Option(attributesArray.getString(R.styleable.SwitchPreference_key))
-
   val switch = findById[Switch](R.id.preference_switch)
-  override val onCheckedChange = EventStream[Boolean]()
-
   var prefKey = Signal[PrefKey[Boolean]]()
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/profile/preferences/views/SwitchPreference.scala]: should this be a var?

File: app/src/main/scala/com/waz/zclient/tracking/GlobalTrackingController.scala [hunk 2]
```diff
@@ -147,9 +147,9 @@ class GlobalTrackingController(implicit inj: Injector, cxt: WireContext, eventCo
           }
         }
       case _ =>
-        trackingEnabled.head.map {
-          case true => send()
-          case _ => //no action
+        trackingEnabled.head.map { tracking =>
+          if (tracking || ignoreTrackingEnabled)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/tracking/GlobalTrackingController.scala]: ignoreTracking not necessary as discussed

**Comment by someoneEsle** [comment]
@red-avtovo my part of the tests looked good

**Comment by red-avtovo** [comment]
Mine is not. No tracking on logout. Weird tracking logs, when tracking is turned off(sent opt-out request). Also need to check with mixpanel

**Comment by someoneEsle** [comment]
Bug 1
User is logged in with account A and B
Account A has tracking disabled, B has tracking enabled
Focus is on account A
User receives notification from account B
User quick-replies to the message
No tracking is generated

**Comment by someoneEsle** [comment]
Bug 2
Logging in is tracked 4 times

**Comment by someoneEsle** [comment]
Bug 3
Registration by phone number is logged 3 times

**Comment by someoneEsle** [comment]
Bug 4
Both properties for team membership should always be displayed.

Question: currently we send the registration.succeeded event after the background picture is chosen. I don't remember who said that it should be tracked after the username has been picked (I might be mistaken). Should it be changed?

**Pull request merged**